packages/eslint-plugin/tests/rules/prefer-optional-chain/prefer-optional-chain.test.ts

```diff
@@ -665,35 +665,189 @@ describe('|| {}', () => {
       'foo ?? {};',
       '(foo ?? {})?.bar;',
       'foo ||= bar ?? {};',
-      // https://github.com/typescript-eslint/typescript-eslint/issues/8380
-      `
-        const a = null;
-        const b = 0;
-        a === undefined || b === null || b === undefined;
-      `,
-      // https://github.com/typescript-eslint/typescript-eslint/issues/8380
-      `
-        const a = 0;
-        const b = 0;
-        a === undefined || b === undefined || b === null;
-      `,
-      // https://github.com/typescript-eslint/typescript-eslint/issues/8380
-      `
-        const a = 0;
-        const b = 0;
-        b === null || a === undefined || b === undefined;
-      `,
-      // https://github.com/typescript-eslint/typescript-eslint/issues/8380
-      `
-        const b = 0;
-        b === null || b === undefined;
-      `,
-      // https://github.com/typescript-eslint/typescript-eslint/issues/8380
-      `
-        const a = 0;
-        const b = 0;
-        b != null && a !== null && a !== undefined;
-      `,
+    ],
+  });
+});
+
+describe('if block with a single statment matches part of the condition', () => {
+  ruleTester.run('prefer-optional-chain', rule, {
+    invalid: [
+      {
+        code: noFormat`if (foo) { foo.bar(); }`,
+        errors: [{ messageId: 'preferOptionalChain', suggestions: null }],
+        options: [{ allowSuggestingOnIfStatements: true }],
+        output: 'foo?.bar();',
+      },
+      {
+        code: noFormat`if (foo) { foo.bar() }`, // Missing semi-colon
```

Comment:
[Style] Personally, I don't find value in a lot of the comments & separation here.
* Rule unit test cases tend to grow organically over time. The comments don't generally age well.
* You can infer from the code anyway what these comments are trying to convey

```suggestion
        code: noFormat`if (foo) { foo.bar() }`,
```